src/praisonai/praisonai/templates/dependency_checker.py

```
@@ -127,6 +144,29 @@ def check_tool(self, tool_name: str) -> Dict[str, Any]:
+        if resolver is None:
+            return None
+        try:
+            if not resolver.has_tool(tool_name):
+                return None
+            # Prefer the resolver's own source attribution when available.
+            try:
+                sources = resolver.list_available_sources()
+                if tool_name in sources:
+                    return sources[tool_name]
+            except Exception:
+                pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)